### PR TITLE
Use std::sort() when it's possible

### PR DESCRIPTION
**Pull request by glx22**
Not replaced QSortT() calls are for C-style arrays.

### CODE REVIEW

**Comment by PeterN** [comment]
Can we use the struct operator() form in some places?

File: src/engine_gui.cpp [hunk 1]
```diff
@@ -325,11 +325,10 @@ void DrawVehicleEngine(int left, int right, int preferred_x, int y, EngineID eng
  */
 void EngList_Sort(GUIEngineList *el, EngList_SortTypeFunction compare)
 {
-	size_t size = el->size();
 	/* out-of-bounds access at the next line for size == 0 (even with operator[] at some systems)
```
Comment (LordAro) [inline src/engine_gui.cpp]: is this true any more?

Comment (glx22) [reply src/engine_gui.cpp]: I don't know, but maybe the comment can go, as anyway sorting less than 2 elements is useless.

**Comment by glx22** [comment]
Hmm yes, it should be possible to use operator() in some places.

File: src/fios.h [hunk 2]
```diff
@@ -107,6 +107,7 @@ struct FiosItem {
 	uint64 mtime;
 	char title[64];
 	char name[MAX_PATH];
+	bool operator() (const FiosItem& da, const FiosItem& db);
```
Comment (michicc) [inline src/fios.h]: Wouldn't `operator <` make even more sense here?

Comment (glx22) [reply src/fios.h]: In this case I'd say yes :)

**Comment by glx22** [comment]
Now using `operator<`. Seems more appropriate.

File: src/fios.h [hunk 3]
```diff
@@ -107,6 +107,7 @@ struct FiosItem {
 	uint64 mtime;
 	char title[64];
 	char name[MAX_PATH];
+	bool operator< (const FiosItem& other);
```
Comment (michicc) [inline src/fios.h]: Missing `const` on the function.

**Review by michicc — APPROVED** [review]

**Pull request merged**